packages/node/src/indexer/indexer.manager.ts

```diff
@@ -170,6 +170,14 @@ export class IndexerManager {
     this.fetchService.latestProcessed(block.block.header.number.toNumber());
     this.prevSpecVersion = block.specVersion;
     if (this.nodeConfig.proofOfIndex) {
+      if (this.blockOffset === undefined) {
```

Comment:
I can see a reason you move the blockOffset initialization here is to skip no operation blocks. If so, the better place is to move it to line 162 and attach it to tx.